### PR TITLE
[Analyzer Plugin] Don't compute disabled diagnostics

### PR DESCRIPTION
**Pull request by timothysteward-wk**
## Motivation
  

Right now when you disable a diagnostic, we still compute the errors for that diagnostic even though the will eventually get filtered out.

## Changes
  
Make it so that when all the diagnostics for a diagnostic contributor have been turned off, skip computing errors for that diagnostic.

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
         - [ ] Turn some diagnostics off and others on. Make sure only the diagnostics that are turned off are ignored.
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tools/analyzer_plugin/test/integration/diagnostics/analysis_options_configuration_test.dart [hunk 1]
```diff
@@ -63,6 +63,15 @@ void main() {
       final errors = await computeErrors();
       expectSeverity(errors, AnalysisErrorSeverity.ERROR);
     });
+
+    // This test ensures that when one of the error codes for a diagnostic contributor is disabled, it doesn't prevent
+    // the other error codes from being computed.
+    test('configured with partially disabled diagnostic contributor', () async {
+      await setUp(yamlContents: yamlPartiallyDisabledContributor);
+      final errors = await computeErrors(sourceCode: badKeysSourceCode);
+      expect(errors, hasLength(1));
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/test/integration/diagnostics/analysis_options_configuration_test.dart:72]: Can we assert somewhere that this code yields two keys when the contributor is not partially or fully disabled, to ensure this test's assumptions don't become incorrect silently if the implementation of these lints changes?

**Review by greglittlefield-wf — CHANGES_REQUESTED** [review]
Just one comment, otherwise looks great!

File: tools/analyzer_plugin/test/integration/diagnostics/analysis_options_configuration_test.dart (hunk 1 above)
Comment (timothysteward-wk) [reply tools/analyzer_plugin/test/integration/diagnostics/analysis_options_configuration_test.dart:72]: Done

**Review by greglittlefield-wf — APPROVED** [review]
+10

**Pull request merged**